src/booking/resolvers/WhitelabelCarRental.js

```
@@ -0,0 +1,70 @@
+// @flow
+
+import { DateTime } from 'luxon';
+import idx from 'idx';
+
+import type { BookingsItem } from '../Booking';
+import type { CarRentalServiceType } from '../types/outputs/services/CarRentalService';
+
+type AncestorType = {|
+  +booking: BookingsItem,
+|};
+
+const lastLocationCode = array => {
+  return array[array.length - 1].arrival.where.code;
+};
+
+const addHours = (date, hours) => {
+  return DateTime.fromJSDate(date, { zone: 'utc' })
+    .plus({ hours })
+    .toJSDate();
+};
+
+export default async ({
+  booking,
+}: AncestorType): Promise<CarRentalServiceType> => {
+  // we are interested in the IATAs at the end of the trips (arrivals only, no transfers)
+  // car rental is then interesting only for the time you are outside of the airport
+
+  const ONE_WEEK = 7 * 24;
+  const locationCodes = new Map();
+
+  if (booking.type === 'BookingOneWay') {
+    const arrivalTime = idx(booking, _ => _.arrival.when.utc) || new Date();
```

Why do you default to `new Date()` if there is no date available? Would it not be better to throw an Error? I probably don't want to be given the option to rent a car with a wrong date -- it would annoy me and cause friction.

---

I don't know what's better here. It returns white label URL at the end so the idea was to construct at least something even if with the wrong dates so the user can correct them (instead of nothing). The dates are going to be mostly wrong anyway because we are setting only the default values here...

---

The user has to click the button and he'll get the opportunity to choose for which part of the trip he needs the car. Then we open the white label with default values. We try to set the default values as he would expect but there are cases where we really don't know. (You arrived to the destination - for how long do you need the car? We say 7 days by default but the real number is unknown.)

Long story short: I don't want to throw an error here. I want to show at least something even though it's maybe wrong.

---

Got it, then it makes sense to me to do what you are doing here. :+1: